sdks/go/container/tools/buffered_logging_test.go

```diff
@@ -166,4 +167,59 @@ func TestBufferedLogger(t *testing.T) {
 			}
 		}
 	})
+
+	t.Run("direct print", func(t *testing.T) {
+		catcher := &logCatcher{}
+		l := &Logger{client: catcher}
+		bl := NewBufferedLogger(l)
+
+		bl.Printf(ctx, "foo %v", "bar")
+
+		received := catcher.msgs[0].GetLogEntries()[0]
+
+		if got, want := received.Message, "foo bar"; got != want {
+			t.Errorf("l.Printf(\"foo %%v\", \"bar\"): got message %q, want %q", got, want)
+		}
+
+		if got, want := received.Severity, fnpb.LogEntry_Severity_DEBUG; got != want {
+			t.Errorf("l.Printf(\"foo %%v\", \"bar\"): got severity %v, want %v", got, want)
+		}
+	})
+
+	t.Run("debug flush at interval", func(t *testing.T) {
+		catcher := &logCatcher{}
+		l := &Logger{client: catcher}
+		interval := time.Duration(5e9)
+		bl := NewBufferedLoggerWithFlushInterval(context.Background(), l, interval)
+
+		messages := []string{"foo", "bar", "baz"}
+
+		for _, message := range messages {
+			// Pause for four seconds before each message
+			time.Sleep(2e9)
```

Comment:
Consider adding a `now` field to the buffered logger that buffered logger uses for "now", defaulting to `time.Now` in the constructors, but can be overridden in these tests.

You'll need to change away from using `time.Since` but this avoids having sleeps in the tests, allowing for fast, deterministic unit tests.

Reply:
Little bit messier in the constructors but it made the unit tests way faster.